index.js

```diff
@@ -89,24 +92,54 @@ router
     "This endpoint is used to determine which metrics (collections) are available to the data source."
   );
 
+const seriesQuery = function(collection, start, end, interval) {
+  const { filterExpression, dateField, valueField, dateExpression,
+          valueExpression } = cfg;
+
+  let filterSnippet = aql.literal(filterExpression
+    ? `FILTER ${filterExpression}`
+    : "");
+
+  let dateSnippet = aql.literal(dateExpression
+    ? `LET d = ${dateExpression}`
+    : `LET d = doc["${dateField}"]`);
+
+  let valueSnippet = aql.literal(valueExpression
+    ? `LET v = ${valueExpression}`
+    : `LET v = doc["${valueField}"]`);
+
+  require("internal").print(`
```

Comment:
This should probably be removed if this is no longer WIP.

Reply:
done